### PR TITLE
ref(tsc): Convert GroupTags to FC and useApiQuery

### PR DESCRIPTION
**Pull request by malwilley**
Ref https://github.com/getsentry/frontend-tsc/issues/2

There already existed a useApiQuery hook for the endpoint used here (`useFetchIssueTags`), so I used that here. Had to slightly change the parameter typings to accommodate the usage.

### CODE REVIEW

File: static/app/views/issueDetails/groupTags.tsx [hunk 1]
```diff
@@ -1,195 +1,167 @@
...
+          start: new Date(breakpoint * 1000).toISOString(),
+          end: new Date(now).toISOString(),
+          durationBaseline: aggregateRange2,
+        }
+      : undefined,
+  });
+
+  // useFetchIssueTags can return two different types of responses, depending on shouldUseTagFacetsEndpoint
+  // This line will convert the response to a common type for rendering
+  const tagList: SimpleTag[] =
```
Comment (malwilley) [inline static/app/views/issueDetails/groupTags.tsx:82]: This is due to `useFetchIssueTags` returning two different types based on whether or not it's a statistical detector issue. I'm not sure I like how a hook is calling two different endpoints, but not going to change that here

File: static/app/views/issueDetails/groupTags.tsx [hunk 2]
```diff
@@ -1,195 +1,167 @@
...
+    organization.features.includes('performance-duration-regression-visible') &&
+    defined(event) &&
+    group.issueType === IssueType.PERFORMANCE_DURATION_REGRESSION;
+
+  const {data, isLoading, isError, refetch} = useFetchIssueTags({
+    orgSlug: organization.slug,
+    groupId: group.id,
+    environment: environments,
+    isStatisticalDetector: shouldUseTagFacetsEndpoint,
+    statisticalDetectorParameters: shouldUseTagFacetsEndpoint
```
Comment (malwilley) [inline static/app/views/issueDetails/groupTags.tsx:70]: @narsaynorath adding you onto the review to make sure I don't break this for statistical detectors!

**Review by narsaynorath — APPROVED** [review]
Changes look good to me! Thanks for looping me in here.

r.e. `useFetchIssueTags` retrieving from multiple endpoints. I agree, it's a little weird for it to be doing that but the data is nearly the same and more flexible for the statistical detector use-case. I think a way forward could have been to update that endpoint but since we had the events facets endpoint available, we went in that direction first before solidifying it into the endpoint

File: static/app/views/issueDetails/groupTags.tsx [hunk 3]
```diff
@@ -1,195 +1,167 @@
...
-        tagList: data
-          .filter(({key}) => key !== 'transaction')
-          .map(sumTagFacetsForTopValues),
-        tagFacets: undefined,
-      });
-    }
+function GroupTags({group, baseUrl, environments, event}: GroupTagsProps) {
+  const organization = useOrganization();
+  const location = useLocation();
+  const now = useMemo(() => Date.now(), []);
```
Comment (scttcper) [inline static/app/views/issueDetails/groupTags.tsx:55]: i've seen us useMemoing the date in a few place but it should be a useRef

File: static/app/views/issueDetails/groupTags.tsx [hunk 4]
```diff
@@ -1,195 +1,167 @@
...
+      : undefined,
+  });
+
+  // useFetchIssueTags can return two different types of responses, depending on shouldUseTagFacetsEndpoint
+  // This line will convert the response to a common type for rendering
+  const tagList: SimpleTag[] =
+    (isTagFacetsResponse(data, shouldUseTagFacetsEndpoint)
+      ? data?.filter(({key}) => key !== 'transaction')?.map(sumTagFacetsForTopValues)
+      : data) ?? [];
+  const alphabeticalTags = (tagList ?? []).sort((a, b) => a.key.localeCompare(b.key));
```
Comment (scttcper) [inline static/app/views/issueDetails/groupTags.tsx:86]: it looks like this is no longer required? or tagList has the wrong type
```suggestion
  const alphabeticalTags = tagList.sort((a, b) => a.key.localeCompare(b.key));
```

File: static/app/views/issueDetails/groupTags.tsx [hunk 5]
```diff
@@ -1,195 +1,167 @@
...
+
+  const {transaction, aggregateRange2, breakpoint} =
+    event?.occurrence?.evidenceData ?? {};
+
+  const shouldUseTagFacetsEndpoint =
+    organization.features.includes('performance-duration-regression-visible') &&
+    defined(event) &&
+    group.issueType === IssueType.PERFORMANCE_DURATION_REGRESSION;
+
+  const {data, isLoading, isError, refetch} = useFetchIssueTags({
```
Comment (scttcper) [inline static/app/views/issueDetails/groupTags.tsx:65]: maybe just add a default here?
```suggestion
  const {data = [], isLoading, isError, refetch} = useFetchIssueTags({
```

**Review by scttcper — APPROVED** [review]

**Pull request merged**